packages/vite/src/node/optimizer/index.ts

```diff
@@ -335,12 +335,10 @@ function needsInterop(
   // ESM dependency's entry chunk into a single default export... detect
   // such cases by checking exports mismatch, and force interop.
   const flatId = flattenId(id) + '.js'
+  const cachePath = normalizePath(path.join(cacheDirOutputPath, flatId))
   let generatedExports: string[] | undefined
   for (const output in outputs) {
-    if (
-      normalizePath(output) ===
-      normalizePath(path.join(cacheDirOutputPath, flatId))
-    ) {
+    if (normalizePath(path.relative(process.cwd(), output)) === cachePath) {
```

Comment:
This doesn't look right to me, it would always result to `false` no?

Reply:
No, it won't always be 'false'. That's why it fixes the problem I encountered by the way.

Reply:
Ah ok. I didn't realize `cacheDirOutputPath` is a relative path. This looks like a harmless change for me then.